core/includes/diff.inc

```diff
@@ -196,8 +196,8 @@ class DiffEngine {
     $edits = array();
     $xi = $yi = 0;
     while ($xi < $n_from || $yi < $n_to) {
-      assert('$yi < $n_to || $this->xchanged[$xi]');
-      assert('$xi < $n_from || $this->ychanged[$yi]');
+      assert($yi < $n_to || $this->xchanged[$xi]);
+      assert($xi < $n_from || $this->ychanged[$yi]);
```

Comment:
Using strings for assertions is deprecated as of PHP 7.2+

https://www.php.net/assert#refsect1-function.assert-parameters